### ISSUE TITLE
crossbeam-channel 0.3.9 triggering Clippy lints

### ISSUE THREAD

**Issue by fizyk20** [crossbeam-rs/crossbeam#404]
The code generated by the `select!` macro in the new version of the crate makes Clippy unhappy:

```
error: `0 as *const _` detected. Consider using `ptr::null()`
   --> src/state_machine.rs:354:13
    |
354 | /             mpmc::select! {
355 | |                 recv(self.network_rx) -> event => self.handle_network_event(event?, outbox),
356 | |                 recv(self.action_rx) -> action => self.handle_action(action?, outbox),
357 | |             }
    | |_____________^
    |
note: lint level defined here
   --> src/lib.rs:104:5
    |
104 |     warnings
    |     ^^^^^^^^
    = note: #[forbid(clippy::zero_ptr)] implied by #[forbid(warnings)]
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#zero_ptr
    = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: calls to `std::mem::drop` with a value that implements Copy. Dropping a copy leaves the original intact.
   --> src/state_machine.rs:354:13
    |
354 | /             mpmc::select! {
355 | |                 recv(self.network_rx) -> event => self.handle_network_event(event?, outbox),
356 | |                 recv(self.action_rx) -> action => self.handle_action(action?, outbox),
357 | |             }
    | |_____________^
    |
    = note: #[deny(clippy::drop_copy)] on by default
note: argument has type [(&dyn crossbeam_channel::select::SelectHandle, usize, *const u8); 2]
   --> src/state_machine.rs:354:13
    |
354 | /             mpmc::select! {
355 | |                 recv(self.network_rx) -> event => self.handle_network_event(event?, outbox),
356 | |                 recv(self.action_rx) -> action => self.handle_action(action?, outbox),
357 | |             }
    | |_____________^
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#drop_copy
    = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
```

As far as I can tell, the first issue could be fixed by replacing `0 as *const u8` with `ptr::null()` here: https://github.com/crossbeam-rs/crossbeam/blob/master/crossbeam-channel/src/select_macro.rs#L767 ; not sure about the second one (the `drop_copy` lint).

**Comment by stjepang** [comment]
> not sure about the second one (the `drop_copy` lint).

I believe the second one can be worked around by replacing `drop(foo);` with `{foo};`

**Comment by Vtec234** [comment]
Should be fixed by #407 .
